common4j/src/main/com/microsoft/identity/common/java/controllers/CommandDispatcher.java

```diff
@@ -491,9 +487,25 @@ private static CommandResult executeCommand(@SuppressWarnings(WarningType.rawtyp
             }
         }
 
+        // set correlation id on Local Authentication Result
+        setCorrelationIdOnResult(commandResult, correlationId);
+        setTelemetryOnResult(commandResult, correlationId);
         return commandResult;
     }
 
+    private static void setTelemetryOnResult(@NonNull final CommandResult commandResult,
```

Comment:
I'm wondering if it's better to create a function that do all the telemetry work in one place 
(attach to command Result AND flush)

Reply:
I'm actually wondering if we should just lean on thread scope to gather broker telemetry rather than attempting to attach it to all these exceptions and result types.  It's probably not time to worry about that, but it would be nicer, I think.

Reply:
Yes, the current design is not great. I want to actually refactor CommandDispatcher and start returning CommandResult out of that...that way only command result contains telemetry. Currently we don't return command result anywhere, it's weird. Refactoring all this CommandDispather is complex and I'll save that for later.

Reply:
Created separate `setOnResultAndFlush` method. Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1636/commits/4f18eff7cf4b93131c2ebd4ec44908688f0efe0b